### PR TITLE
Add the doc on Gateway

### PR DESCRIPTION
**Pull request by azatsarynnyy**
## What does this pull request change?
Adds the doc describes how to connect the IntelliJ Thin Client to a remote DevWorkspace over the JetBrains Gateway application.

## What issues does this pull request fix or reference?
https://issues.redhat.com/browse/CRW-5750

## Specify the version of the product this pull request applies to
Eclipse Che 7.86
Not for Dev Spaces yet.

## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 1]
```diff
@@ -0,0 +1,52 @@
+[id="idea-ultimate"]
+= JetBrains IntelliJ IDEA Ultimate Edition (over JetBrains Gateway)
+
+Eclipse Che supports connecting your local IntelliJ IDE to a running Che workspace.
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc]:
```suggestion
{prod} supports connecting your local IntelliJ IDE to a running {prod-short} workspace over JetBrains Gateway.
```

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+.Prerequisites
+
+. Install the JetBrains Gateway application from https://www.jetbrains.com/remote-development/gateway/
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc]:
```suggestion
. link:https://www.jetbrains.com/remote-development/gateway/[The JetBrains Gateway application] is installed.
```

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+. Install the Gateway provider for OpenShift Dev Spaces from https://plugins.jetbrains.com/plugin/24234-openshift-dev-spaces
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc]:
```suggestion
. link:https://plugins.jetbrains.com/plugin/24234-openshift-dev-spaces[The Gateway provider for OpenShift Dev Spaces] is installed.
```

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 4]
```diff
@@ -0,0 +1,52 @@
+[id="idea-ultimate"]
+= JetBrains IntelliJ IDEA Ultimate Edition (over JetBrains Gateway)
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc]:
```suggestion
= Connecting JetBrains IntelliJ IDEA Ultimate Edition to a {prod-short} workspace
```

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+Eclipse Che supports connecting your local IntelliJ IDE to a running Che workspace.
+
+.Prerequisites
+
+. Install the JetBrains Gateway application from https://www.jetbrains.com/remote-development/gateway/
+
+. Install the Gateway provider for OpenShift Dev Spaces from https://plugins.jetbrains.com/plugin/24234-openshift-dev-spaces
+
+. In your local terminal, log in to your OpenShift server with the OpenShift Client. `oc login` command establishes the authenticated session and saves the connection information to the configuration file which is read by the Gateway provider for OpenShift Dev Spaces.
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc]:
```suggestion
. In your local terminal, you are logged in to your OpenShift server with the OpenShift Client.
+
[NOTE]
====
The `oc login` command establishes the authenticated session and saves the connection information to the configuration file which is read by the Gateway provider for OpenShift Dev Spaces.
====

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+.Procedure of connecting to a new workspace
+
+. When creating your workspace on Che Dashboard, choose `IntelliJ IDEA Ultimate (desktop)` editor:
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc]:
```suggestion
. Create a workspace on the {prod-short} Dashboard and choose `IntelliJ IDEA Ultimate (desktop)` editor:
```

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
++
+image::editor-idea-iu.png[IntelliJ IDEA Ultimate on Dashboard,link="{imagesdir}/editor-idea-iu.png"]
+
+. Once the workspace runs, the Dashboard will ask if you want to open your local JetBrains Gateway application:
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc]:
```suggestion
. Wait for the prompt to open your local JetBrains Gateway application to appear:
```

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
++
+image::open-gateway-prompt.png[Open Gateway prompt,link="{imagesdir}/open-gateway-prompt.png"]
+
+Clicking the `Open Gateway` button will start your local JetBrains Client application connected to your Dev Spaces workspace:
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc]:
```suggestion
Click the `Open Gateway` button to start your local JetBrains Client application connected to your Dev Spaces workspace:
```

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+.Prerequisites
+
+. Install the JetBrains Gateway application from https://www.jetbrains.com/remote-development/gateway/
+
+. Install the Gateway provider for OpenShift Dev Spaces from https://plugins.jetbrains.com/plugin/24234-openshift-dev-spaces
+
+. In your local terminal, log in to your OpenShift server with the OpenShift Client. `oc login` command establishes the authenticated session and saves the connection information to the configuration file which is read by the Gateway provider for OpenShift Dev Spaces.
+
+.Procedure of connecting to a new workspace
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc]:
```suggestion
.Connecting JetBrains IntelliJ IDEA Ultimate Edition to a new workspace

.Procedure
```

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+. Once the workspace runs, the Dashboard will ask if you want to open your local JetBrains Gateway application:
++
+image::open-gateway-prompt.png[Open Gateway prompt,link="{imagesdir}/open-gateway-prompt.png"]
+
+Clicking the `Open Gateway` button will start your local JetBrains Client application connected to your Dev Spaces workspace:
+
+image::gateway-connecting.png[Connecting to remote host,link="{imagesdir}/gateway-connecting.png"]
+
+.Procedure of connecting to an existing workspace
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc]:
```suggestion
.Connecting JetBrains IntelliJ IDEA Ultimate Edition to an existing workspace
```

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+There are two options to connect to an already existing workspace:
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc]:
```suggestion
There are two options to connect the local IntelliJ IDE to an existing workspace:
```

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+
+* from the Che Dashboard
+* from the Gateway application
+
+The simplest way to connect to an existing workspace is to start a workspace from the Che Dashboard and click the `Open Gateway` button.
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc]:
```suggestion
The simplest way to connect to an existing workspace is to start a workspace from the {prod-short} Dashboard and click the `Open Gateway` button.
```

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+Another option to connect is using the Gateway application. It allows connecting without accessing the Che Dashboard.
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc]:
```suggestion
However, with the Gateway application, you can connect the IntelliJ IDE to your workspace without accessing the {prod-short} Dashboard.
```

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
+. Open the Gateway app and click `Connect to Dev Spaces`:
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc:47]:
```suggestion
.Procedure

. Open the Gateway app and click `Connect to Dev Spaces`:
```

File: modules/end-user-guide/pages/idea-ultimate.adoc [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+
+. On the next page, provide the parameters to connect to the OpenShift API server and click the `Check Connection and Continue` button:
++
+image::gateway-connect.png[Connecting to OpenShift API server,link="{imagesdir}/gateway-connect.png"]
+
+. Choose your workspace and click the `Connect` button:
++
+image::gateway-select-ws.png[Connecting to OpenShift API server,link="{imagesdir}/gateway-select-ws.png"]
+
+Your local Gateway application will run the JetBrains Client and connect to the workspace.
```
Comment (deerskindoll) [inline modules/end-user-guide/pages/idea-ultimate.adoc]:
```suggestion

.Verification

. Your local Gateway application is running the JetBrains Client and connects to the workspace.
+
```

**Review by deerskindoll — COMMENTED** [review]
I moved things around a bit to make the guide easier to follow.

**Comment by azatsarynnyy** [comment]
@deerskindoll thanks a lot for suggesting your improvements to my PR. I applied all of them.

**Comment by deerskindoll** [comment]
> @deerskindoll thanks a lot for suggesting your improvements to my PR. I applied all of them.

you're welcome! thank you for taking the suggestions into consideration! 

P.S. did you know there's a sake emoji 🍶 in github? I discovered it accidentally when writing this response. also: a safety pin 🧷 !

**Review by deerskindoll — APPROVED** [review]
lgtm

**Comment by azatsarynnyy** [comment]
> P.S. did you know there's a sake emoji 🍶 in github? I discovered it accidentally when writing this response. also: a safety pin 🧷 !

@deerskindoll  it's a nice accidental findings 
I like it )

**Comment by ibuziuk** [comment]
@deerskindoll please, make sure that those docs are NOT part of the downstream Dev Spaces 3.14 documentation

**Comment by deerskindoll** [comment]
> @deerskindoll please, make sure that those docs are NOT part of the downstream Dev Spaces 3.14 documentation

sure!

**Comment by deerskindoll** [comment]
added to downstream inject-collector-collector.js

**Pull request merged**